spec/models/crop_spec.rb

```diff
@@ -328,6 +328,8 @@
   context "search" do
     before :each do
       @mushroom = FactoryGirl.create(:crop, :name => 'mushroom')
+      @mushroom.__elasticsearch__.index_document
+      sleep 1
```

Comment:
Good spot! Yes, definitely try to make this test run synchronously, or put
the assertion in a callback, or something.
On Jan 16, 2015 8:41 PM, "Taylor Griffin" <notifications@github.com> wrote:

> In spec/models/crop_spec.rb
> <https://github.com/Growstuff/growstuff/pull/648#discussion-diff-23107889>
> :
>
> > @@ -328,6 +328,8 @@
> >    context "search" do
> >      before :each do
> >        @mushroom = FactoryGirl.create(:crop, :name => 'mushroom')
> > +      @mushroom.__elasticsearch__.index_document
> > +      sleep 1
>
> This sleep is here to manage the asynchronicity of the indexing, right?
> Might be better to handle this without sleep, because this can sometimes
> cause spuriously failing tests. I'm not sure if this is possible without
> doing a little research. If it turns out to be a big hassle then we can
> probably live with this since indexing one record should be super fast.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Growstuff/growstuff/pull/648/files#r23107889>.
>